pkg/block/fetcher.go

```diff
@@ -286,12 +379,30 @@ type response struct {
 	incompleteView bool
 }
 
-func (f *BaseFetcher) fetchMetadata(ctx context.Context) (interface{}, error) {
+func (r response) metasCopy() map[ulid.ULID]*metadata.Meta {
+	metas := make(map[ulid.ULID]*metadata.Meta, len(r.metas))
+	for id, m := range r.metas {
+		metas[id] = m
+	}
+	return metas
+}
+
+// deletionMarksCopyForFilter makes a copy of deletion marks map, suitable for passing to Filter method.
+func (r response) deletionMarksCopyForFilter() map[ulid.ULID]*metadata.DeletionMark {
```

Comment:
Looks a bit weird to me calling it `ForFilter()`. The usage we're going to do with a copy of deletion marks shouldn't be reflected in the function name IMO.